src/m365/aad/commands/administrativeunit/administrativeunit-member-add.ts

```diff
@@ -0,0 +1,166 @@
+      },
+      {
+        option: "--groupName [groupName]"
+      },
+      {
+        option: "--deviceId [deviceId]"
+      },
+      {
+        option: "--deviceName [deviceName]"
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.administrativeUnitId && !validation.isValidGuid(args.options.administrativeUnitId as string)) {
+          return `${args.options.administrativeUnitId} is not a valid GUID`;
+        }
+
+        if (args.options.userId && !validation.isValidGuid(args.options.userId as string)) {
+          return `${args.options.userId} is not a valid GUID`;
+        }
+
+        if (args.options.groupId && !validation.isValidGuid(args.options.groupId as string)) {
+          return `${args.options.groupId} is not a valid GUID`;
+        }
+
+        if (args.options.deviceId && !validation.isValidGuid(args.options.deviceId as string)) {
+          return `${args.options.deviceId} is not a valid GUID`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({ options: ['administrativeUnitId', 'administrativeUnitName'] });
+    this.optionSets.push({ options: ['userId', 'userName', 'groupId', 'groupName', 'deviceId', 'deviceName'] });
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    let administrativeUnitId = args.options.administrativeUnitId;
+    let memberType;
+    let memberId;
+
+    try {
+      if (args.options.administrativeUnitName) {
+        administrativeUnitId = (await aadAdministrativeUnit.getAdministrativeUnitByDisplayName(args.options.administrativeUnitName)).id!;
+      }
+
+      if (args.options.userId || args.options.userName) {
+        memberType = 'users';
+        memberId = args.options.userId;
+        if (args.options.userName) {
+          memberId = await aadUser.getUserIdByUpn(args.options.userName);
+        }
+      }
+      else if (args.options.groupId || args.options.groupName) {
+        memberType = 'groups';
+        memberId = args.options.groupId;
+        if (args.options.groupName) {
+          memberId = await aadGroup.getGroupIdByDisplayName(args.options.groupName);
+        }
+      }
+      else if (args.options.deviceId || args.options.deviceName) {
+        memberType = 'devices';
+        memberId = args.options.deviceId;
+        if (args.options.deviceName) {
+          memberId = (await aadDevice.getDeviceByDisplayName(args.options.deviceName)).id;
+        }
+      }
+
+      const requestOptions: CliRequestOptions = {
+        url: `${this.resource}/v1.0/directory/administrativeUnits/${administrativeUnitId}/members/$ref`,
+        headers: {
+          'accept': 'application/json;odata.metadata=none'
+        },
+        data: {
+          "@odata.id": `https://graph.microsoft.com/v1.0/${memberType}/${memberId}`
+        }
+      };
+
+      await request.post(requestOptions);
```

Comment:
Does the endpoint have some output? If so, let's log that

Reply:
This endpoint returns `204 No Content`.